slasher/beaconclient/validator_retrieval.go

```diff
@@ -0,0 +1,56 @@
+package beaconclient
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"go.opencensus.io/trace"
+)
+
+// FindOrGetPublicKeys gets public keys from cache or request validators public
+// keys from a beacon node via gRPC.
+func (bs *Service) FindOrGetPublicKeys(
+	ctx context.Context,
+	validatorIndices []uint64,
+) (map[uint64][]byte, error) {
+	ctx, span := trace.StartSpan(ctx, "beaconclient.FindOrGetPublicKeys")
+	defer span.End()
+
+	validators := make(map[uint64][]byte, len(validatorIndices))
+	notFound := 0
+	for _, validatorIdx := range validatorIndices {
+		pub, exists := bs.publicKeyCache.Get(validatorIdx)
+		if exists {
+			validators[validatorIdx] = pub
+			continue
+		}
+		validatorIndices[notFound] = validatorIdx
+		notFound++
```

Comment:
What are you doing here with the `notFound`?

Reply:
removing found elements from slice

Reply:
added comment